### PR TITLE
feat(Isolation): Added isolation support

### PR DESCRIPTION
**Pull request by zhenlei520**

### CODE REVIEW

File: src/Data/Masa.BuildingBlocks.Data.UoW/IUnitOfWorkManager.cs [hunk 1]
```diff
@@ -1,7 +1,13 @@
 ﻿namespace Masa.BuildingBlocks.Data.UoW;
+
 public interface IUnitOfWorkManager
 {
-    IUnitOfWork CreateDbContext();
+    /// <summary>
+    ///
```
Comment (doddgu) [inline src/Data/Masa.BuildingBlocks.Data.UoW/IUnitOfWorkManager.cs:6]: ?

File: src/Isolation/Masa.BuildingBlocks.Isolation/ITenantParserProvider.cs [hunk 2]
```diff
@@ -0,0 +1,7 @@
+﻿namespace Masa.BuildingBlocks.Isolation;
+public interface ITenantParserProvider
+{
+    string Name { get;}
+
+    Task<bool> ExecuteAsync(IServiceProvider serviceProvider);
```
Comment (doddgu) [inline src/Isolation/Masa.BuildingBlocks.Isolation/ITenantParserProvider.cs:6]: Why environment called resolved, and this called execute?

File: src/Isolation/Masa.BuildingBlocks.Isolation/MultiTenant/Tenant.cs [hunk 3]
```diff
@@ -0,0 +1,11 @@
+﻿namespace Masa.BuildingBlocks.Isolation.MultiTenant;
+public class Tenant
+{
+    public string Id { get; set; }
+
+    public Tenant(object id) : this(id.ToString() ?? String.Empty)
```
Comment (doddgu) [inline src/Isolation/Masa.BuildingBlocks.Isolation/MultiTenant/Tenant.cs:6]: id?.ToString() ?? throw exception xxx

File: src/Isolation/Masa.BuildingBlocks.Isolation/Options/DbConnectionOptions.cs [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+    public string TenantId { get; set; }
+
+    public string Environment { get; set; }
+
+    public string ConnectionString { get; set; }
+
+    /// <summary>
+    /// Used to control the configuration with the highest score when multiple configurations are satisfied. The default score is 100
+    /// </summary>
+    public int Score { get; set; } = 100;
```
Comment (doddgu) [inline src/Isolation/Masa.BuildingBlocks.Isolation/Options/DbConnectionOptions.cs:13]: If there has the same score, what's the control logic?

Comment (zhenlei520) [reply src/Isolation/Masa.BuildingBlocks.Isolation/Options/DbConnectionOptions.cs:13]: If the scores are the same, choose the first one in the order of the Isolations configuration

File: src/Isolation/Masa.BuildingBlocks.Isolation/MultiTenant/Tenant.cs (hunk 3 above)
Comment (zhenlei520) [reply src/Isolation/Masa.BuildingBlocks.Isolation/MultiTenant/Tenant.cs:6]: The original intention here is to allow the tenant to be manually set to be empty, and adjust the tenant id through code to achieve the purpose of changing the DbContext and skipping the tenant filtering. But after thinking about it carefully, I hope that changing the DbContext can be done by creating a new DbContext through IUnitOfWorkManager, and skipping tenant filtering can be done through IDataFilter. The design here is not suitable.

File: src/Data/Masa.BuildingBlocks.Data.UoW/IUnitOfWorkManager.cs (hunk 1 above)
Comment (zhenlei520) [reply src/Data/Masa.BuildingBlocks.Data.UoW/IUnitOfWorkManager.cs]: preocessed

File: src/Isolation/Masa.BuildingBlocks.Isolation/ITenantParserProvider.cs (hunk 2 above)
Comment (zhenlei520) [reply src/Isolation/Masa.BuildingBlocks.Isolation/ITenantParserProvider.cs]: processed

**Review by doddgu — APPROVED** [review]
LGTM

**Pull request merged**